docs/templates/model-serving/predictor.md

```diff
@@ -0,0 +1,167 @@
+
+### Custom predictor
+
+Depending on the model server and serving tool used in the deployment, users can provide their own python script to load the model and make predictions.
+
+??? info "Show supported custom predictors"
+
+    | Serving tool | Model server       | Custom predictor script |
+    | ------------ | ------------------ | ----------------------- |
+    | Docker       | Flask              | ❌                      |
+    |              | Tensorflow Serving | ✅                      |
+    | Kubernetes   | Flask              | ❌                      |
+    |              | Tensorflow Serving | ✅                      |
+    | KServe       | Flask              | ✅                      |
+    |              | Tensorflow Serving | ✅                      |
+
+To configure a custom predictor, users must provide a python script implementing the following class.
+
+=== "Python"
+
+    ``` python
+    class Predict(object):
+
+        def __init__(self):
+            """ Initialization code goes here:
+                - Download the model artifact
+                - Load the model
+            """
+            pass
+
+        def predict(self, inputs):
+            """ Serve predictions using the trained model"""
+            pass
+    ```
+
+The path to this script has to be provided when calling `deploy()` or `create_predictor()` methods. Find more details in the [Predictor Reference](predictor_api.md).
+
+See examples of custom predictor scripts in the serving [example notebooks](https://github.com/logicalclocks/hops-examples/blob/master/notebooks/ml/serving).
+
+### Configuration
+
+#### Resource allocation
+
+Depending on the combination of serving tool used to deploy the a model, resource allocation can be configured at different levels. While predictors on Docker containers are fixed in resources, using Kubernetes or KServe allows a better exploitation of the resources available in the platform.
+
+??? info "Show supported resource allocation configuration"
+
+    | Serving tool | Component   | Resources                   |
+    | ------------ | ----------- | --------------------------- |
+    | Docker       | Predictor   | Fixed                       |
+    |              | Transformer | ❌                          |
+    | Kubernetes   | Predictor   | Minimum resources           |
+    |              | Transformer | ❌                          |
+    | KServe       | Predictor   | Minimum / maximum resources |
+    |              | Transformer | Minimum / maximum resources |
+
+To learn how to configure resource allocation for a deployment, see [Resources Reference](resources_api.md)
+
+#### Inference logger
+
+Once a model is deployed and starts making predictions as inference requests arrive, logging model inputs and predictions become essential to monitor the health of the model and take action when it degrades over time.
```

Comment:
becomes essential

 when it degrades -> if the model's performance degrades over time